src/components/presentational/TimeSeriesChart/TimeSeriesChart.tsx

```diff
@@ -0,0 +1,331 @@
+import React, { useContext } from 'react'
+import { add, addDays, addMonths, format, isToday } from 'date-fns'
+import { CardTitle, LayoutContext, LoadingIndicator } from '..'
+import { Area, AreaChart, Bar, BarChart, CartesianGrid, Cell, Line, LineChart, ReferenceLine, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import "./TimeSeriesChart.css"
+import { AxisDomain } from 'recharts/types/util/types'
+import { ColorDefinition, resolveColor } from '../../../helpers/colors'
+import getDaysInMonth from 'date-fns/getDaysInMonth'
+import { ceil } from 'lodash'
+import addHours from 'date-fns/addHours'
+import startOfMonth from 'date-fns/startOfMonth'
+
+export interface TimeSeriesChartProps {
+    title?: string
+    intervalType?: "Week" | "Month" | "6Month" | "Day",
+    intervalStart: Date,
+    data: any[] | undefined,
+    expectedDataInterval?: Duration,
+    dataKeys?: string[],
+    chartHasData: boolean,
+    tooltip: ({ active, payload, label }: any) => React.JSX.Element | null,
+    chartType: "Line" | "Bar" | "Area"
+    options?: LineChartOptions | BarChartOptions | AreaChartOptions
+    innerRef?: React.Ref<HTMLDivElement>
+}
+
+export interface LineChartOptions {
+    lineColor?: ColorDefinition | ColorDefinition[],
+    connectNulls?: boolean,
+    domainMin?: number | "Auto"
+}
+
+export interface BarChartOptions {
+    barColor?: ColorDefinition | ColorDefinition[],
+    thresholds?: BarChartThreshold[]
+}
+
+export interface BarChartThreshold {
```

Comment:
I would prefer not to.  I think having such a generic name isn't helpful to have as an export when it's use is pretty narrow to its usage in this class.

Reply:
I'm indifferent to what it is named if Threshold is too generic. 
ChartThresholds is another option. 

Given a threshold lines are shown in both bar and line, it is confusing for the name to be bar chart specific.

Reply:
This has been adjusted slightly in the last update.  There's still a BarChartThreshold for the DailyDataChart, but there is a more generically named one that the SurveyAnswer and TimeSeries chart use (which eliminates the word Bar from both the interface name and the property inside the interface).  Keeping the existing one around serves to not break the current DailyDataChart interface which would technically require a major version number bump that I'd like to avoid.